### DISCUSSION TITLE
Use of declared argument in get_post_login_redirect function

### DISCUSSION THREAD

**Discussion by TheDavidFactor** [pallets-eco/flask-security#805] [Q&A]
I see `declared` is passed through `get_post_login_redirect` to `get_post_action_redirect` where it is inserted into the first position of the `urls` list if it's provided. However I don't see any call to `get_post_login_redirect` that sets `declared` so it appears to be unused.

I need to be able to override the login redirect from a custom login form `validate` method on occasion. I'm doing this by setting a value for `security_post_login_view` in the session since that is checked first in `find_redirect` but it would be nice to be able to preempt the arg and form next values as `declared` does.

Is there somewhere I could hook into `declared` that I'm not seeing? Or would I need to subclass or monkey patch part of Flask-Security-Too to get access to setting `declared`?

**Comment by jwag956** [comment]
That argument - and not being used - has been there forever... Yours is the first useful use case I have heard about - let me think about that a bit. I suppose a not too bad hack would be to change 'next' on the request and let the normal logic take its course.
